neqo-transport/src/cc.rs

```diff
@@ -126,31 +128,40 @@ impl CongestionControl {
                 continue;
             }
 
-            if self.congestion_window < self.ssthresh {
-                self.congestion_window += pkt.size;
-                qinfo!([self], "slow start");
-                qlog::congestion_state_updated(
-                    &mut self.qlog,
-                    &mut self.qlog_curr_cong_state,
-                    CongestionState::SlowStart,
-                );
-            } else {
-                self.congestion_window += (MAX_DATAGRAM_SIZE * pkt.size) / self.congestion_window;
-                qinfo!([self], "congestion avoidance");
-                qlog::congestion_state_updated(
-                    &mut self.qlog,
-                    &mut self.qlog_curr_cong_state,
-                    CongestionState::CongestionAvoidance,
-                );
-            }
-            qlog::metrics_updated(
+            self.acked_bytes += pkt.size;
+        }
+        qtrace!([self], "ACK received, acked_bytes = {}", self.acked_bytes);
+
+        // Slow start, up to the slow start threshold.
+        if self.congestion_window < self.ssthresh {
+            let increase = min(self.ssthresh - self.congestion_window, self.acked_bytes);
+            self.congestion_window += increase;
+            self.acked_bytes -= increase;
+            qinfo!([self], "slow start += {}", increase);
+            qlog::congestion_state_updated(
                 &mut self.qlog,
-                &[
-                    QlogMetric::CongestionWindow(self.congestion_window),
-                    QlogMetric::BytesInFlight(self.bytes_in_flight),
-                ],
+                &mut self.qlog_curr_cong_state,
+                CongestionState::SlowStart,
             );
         }
+        // Congestion avoidance, above the slow start threshold.
```

Comment:
We need `else` here:
```
if self.congestion_window < self.ssthresh {
...
} else { --> this is missing
...
```

Reply:
That too was intentional.  If we hit ssthresh, the bytes up to that limit will count 1:1, the bytes after will be counted toward the next max_datagram_size increase.